src/Node/Expression/AssignNameExpression.php

```diff
@@ -15,8 +15,9 @@
 use Twig\Compiler;
 use Twig\Error\SyntaxError;
 use Twig\Node\Expression\Variable\AssignContextVariable;
+use Twig\Node\Expression\Variable\ContextVariable;
 
-class AssignNameExpression extends NameExpression
+class AssignNameExpression extends ContextVariable
```

Comment:
This is the main changes that allows us to remove any `instanceof NameExpression` in the code (as `NameExpression` is deprecated).